tests/tpch/test_duckdb.py

```diff
@@ -312,3 +312,673 @@ def _():
+                and l_receiptdate >= date '1994-01-01'
+                and l_receiptdate < date '1994-01-01' + interval '1' year
+            group by
+                l_shipmode
+            order by
+                l_shipmode
+            """
+        ).arrow()
+
+    run(_)
+
+
+def test_query_13(run, connection, dataset_path):
+    def _():
+        connection().execute(
+            f"""
+            with customer as (select * from read_parquet('{dataset_path}customer/*.parquet')),
+                 orders as (select * from read_parquet('{dataset_path}orders/*.parquet'))
+
+            select
+                c_count, count(*) as custdist
+            from (
+                select
+                    c_custkey,
+                    count(o_orderkey)
+                from
+                    customer left outer join orders on
+                    c_custkey = o_custkey
+                    and o_comment not like '%special%requests%'
+                group by
+                    c_custkey
+                )as c_orders (c_custkey, c_count)
+            group by
+                c_count
+            order by
+                custdist desc,
+                c_count desc
+            """
+        ).arrow()
+
+    run(_)
+
+
+def test_query_14(run, connection, dataset_path):
+    def _():
+        connection().execute(
+            f"""
+            with lineitem as (select * from read_parquet('{dataset_path}lineitem/*.parquet')),
+                 part as (select * from read_parquet('{dataset_path}part/*.parquet'))
+
+            select
+                round(100.00 * sum(case
+                    when p_type like 'PROMO%'
+                        then l_extendedprice * (1 - l_discount)
+                    else 0
+                end) / sum(l_extendedprice * (1 - l_discount)), 2) as promo_revenue
+            from
+                lineitem,
+                part
+            where
+                l_partkey = p_partkey
+                and l_shipdate >= date '1995-09-01'
+                and l_shipdate < date '1995-09-01' + interval '1' month
+            """
+        ).arrow()
+
+    run(_)
+
+
+def test_query_15(run, connection, dataset_path):
+    def _():
+        connection().execute(
+            f"""
+            with lineitem as (select * from read_parquet('{dataset_path}lineitem/*.parquet'))
+
+            create or replace temporary view revenue (supplier_no, total_revenue) as
+                select
+                    l_suppkey,
+                    sum(l_extendedprice * (1 - l_discount))
+                from
+                    lineitem
+                where
+                    l_shipdate >= date '1996-01-01'
+                    and l_shipdate < date '1996-01-01' + interval '3' month
+                group by
+                    l_suppkey
+            """
+        )
+
+        connection.execute(
+            f"""
+            with supplier as (select * from read_parquet('{dataset_path}supplier/*.parquet'))
```

Comment:
untested, but think something like this will work

```suggestion
            with lineitem as (select * from read_parquet('{dataset_path}lineitem/*.parquet')),
                     revenue as (
                         select
                             l_suppkey as supplier_no,
                             sum(l_extendedprice * (1 - l_discount)) as total_revenue
                         from
                             lineitem
                         where
                             l_shipdate >= date '1996-01-01'
                             and l_shipdate < date '1996-01-01' + interval '3' month
                         group by
                             l_suppkey
                     ),
                    supplier as (select * from read_parquet('{dataset_path}supplier/*.parquet'))
```

Reply:
The spec wants to see a view, so I've just removed the with statement.